<launch>
    <!-- 启动路径规划节点 -->
    <group ns="iris_0">
        <node name="ship_painter_node" pkg="ship_painter" type="ship_painter_node" output="screen">
            <param name="model_path" value="$(find ship_painter)/models/test.stl"/>
            <param name="offset_distance" value="0.5"/>
            <param name="flight_height" value="5.0"/>
            <param name="frame_id" value="world"/>
            
            <!-- 添加模型位置参数，对应test1中spawn的位置 -->
            <param name="model_x" value="5.0"/>
            <param name="model_y" value="0.0"/>
            <param name="model_z" value="0.25"/>
            
            <!-- 完全重映射MAVROS话题 -->
            <remap from="mavros/state" to="mavros/state"/>
            <remap from="mavros/local_position/pose" to="mavros/local_position/pose"/>
            <remap from="mavros/setpoint_position/local" to="mavros/setpoint_position/local"/>
            <remap from="mavros/set_mode" to="mavros/set_mode"/>
            <remap from="mavros/cmd/arming" to="mavros/cmd/arming"/>
        </node>
    </group>
    
    <!-- 启动 Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ship_painter)/rviz/ship_painter.rviz"/>
    <node pkg="tf" type="static_transform_publisher" name="world_to_map" args="0 0 0 0 0 0 world map 100"/>
</launch>